src/metabase/sync/analyze/classifiers/category.clj

```diff
@@ -9,22 +9,29 @@
             [schema.core :as s]))
 
 
-(s/defn ^:private cannot-be-category? :- s/Bool
-  [base-type :- su/FieldType]
-  (or (isa? base-type :type/DateTime)
-      (isa? base-type :type/Collection)))
+(def ^:const ^:Integer category-string-length-threshold
+  "Maximum average string length for a field to be considered a category."
+  20)
+
+(s/defn ^:private category-candidate? :- s/Bool
+  [{:keys [base_type special_type database_type]}]
+  (and (nil? special_type)
+       (or (isa? base_type :type/Integer)
+           (isa? base_type :type/Text))
+       (not= database_type "metric")))
```

Comment:
This is Druid-specific, I'm assuming? It's probably a good idea to but a short comment there explaining that so it's not confusing to other people reading the code

Reply:
Yes. Comment added. Tnx.